src/user-input-sequence.ts

```diff
@@ -0,0 +1,82 @@
+import * as sourcegraph from 'sourcegraph'
+import { PullRequestBundle, SourceTree } from './types'
+
+export async function bundlePullRequests(sourceTrees: SourceTree[]): Promise<PullRequestBundle[]> {
+    async function getConfValueOrPrompt(key: string, prompt: string, placeholder?: string): Promise<string> {
+        if (!sourcegraph.app.activeWindow) {
+            throw new Error('No active window')
+        }
+
+        const value =
+            sourcegraph.configuration.get().value[key] !== undefined
+                ? sourcegraph.configuration.get().value[key]
+                : await sourcegraph.app.activeWindow.showInputBox({
+                      prompt,
+                      value: placeholder,
+                  })
+        if (value === undefined) {
+            throw new Error('Pull request canceled.')
+        }
+        return value
+    }
+
+    const bundles: PullRequestBundle[] = []
+
+    for (const { owner, repo, sourceFiles } of sourceTrees) {
+        if (!sourcegraph.app.activeWindow) {
+            throw new Error('No active window')
+        }
+        await sourcegraph.app.activeWindow.showMessage(`PR for ${owner}/${repo}`)
+
+        const gitHubAccessToken = await getConfValueOrPrompt('pr.gitHubAccessToken', 'GitHub Access Token:')
+        const subject = await getConfValueOrPrompt('pr.title', 'Pull Request Title:', 'codemod:')
+        const description = await getConfValueOrPrompt('pr.description', 'Pull Request Description:', '...')
+        const commitMessage = await getConfValueOrPrompt(
+            'pr.commitMessage',
+            'Commit Message:',
+            sourcegraph.configuration.get().value['pr.codemodQuery']
+        )
+        const authorName = await getConfValueOrPrompt('pr.authorName', 'Author Name:')
+        const authorEmail = await getConfValueOrPrompt('pr.authorEmail', 'Author Email:')
+        const sourceBranch = await getConfValueOrPrompt(
+            'pr.sourceBranch',
+            'This feature branch name (e.g., my-codemod-branch):'
+        )
+
+        // Don't prompt the branch this PR is made against: assume the default is 'master' if not specified in config.
+        const destinationBranch = sourcegraph.configuration.get().value['pr.destinationBranch'] || 'master'
```

Comment:
Note that `master` is not necessarily the default branch. You can get this from the GitHub API from a Repository's `default_branch` property

Reply:
Good point, tracking in #5